src/auth/src/credentials/mds.rs

```diff
@@ -175,6 +180,77 @@ impl Builder {
         self.scopes = Some(scopes.into_iter().map(|s| s.into()).collect());
         self
     }
+    /// Configure the retry policy for fetching tokens.
+    ///
+    /// The authentication library can automatically retry operations that fail. The
+    /// retry policy controls how to handle retries, and sets limits on the
+    /// number of attempts or the total time spent retrying.
+    ///
+    /// ```
+    /// # use google_cloud_auth::credentials::mds::Builder;
+    /// # use gax::retry_policy;
+    /// # use gax::retry_policy::RetryPolicyExt;
+    /// # tokio_test::block_on(async {
+    /// let credentials = Builder::default()
+    ///     .with_retry_policy(retry_policy::AlwaysRetry.with_attempt_limit(3))
+    ///     .build();
+    /// # });
+    /// ```
+    pub fn with_retry_policy<V: Into<RetryPolicyArg>>(mut self, v: V) -> Self {
+        self.retry_builder = self.retry_builder.with_retry_policy(v.into().0);
+        self
+    }
+
+    /// Configure the retry backoff policy.
+    ///
+    /// The authentication library can automatically retry operations that fail. The
+    /// backoff policy controls how long to wait in between retry attempts.
+    ///
+    /// ```
+    /// # use google_cloud_auth::credentials::mds::Builder;
+    /// # use gax::exponential_backoff::ExponentialBackoffBuilder;
+    /// # use std::time::Duration;
+    /// # tokio_test::block_on(async {
+    /// let policy = ExponentialBackoffBuilder::new()
+    ///     .with_initial_delay(Duration::from_millis(100))
+    ///     .with_maximum_delay(Duration::from_secs(5))
+    ///     .with_scaling(4.0)
+    ///     .build().expect("well-known policy values should succeed");
+    /// let credentials = Builder::default()
+    ///     .with_backoff_policy(policy)
+    ///     .build();
+    /// # });
+    /// ```
+    pub fn with_backoff_policy<V: Into<BackoffPolicyArg>>(mut self, v: V) -> Self {
+        self.retry_builder = self.retry_builder.with_backoff_policy(v.into().0);
+        self
+    }
+
+    /// Configure the retry throttler.
+    ///
+    /// Advanced applications may want to configure a retry throttler to
+    /// [Address Cascading Failures] and when [Handling Overload] conditions.
+    /// The authentication library throttles its retry loop, using a policy to
+    /// control the throttling algorithm. Use this method to fine tune or
```

Comment:
Same here, we do not have a default throttler. Customer has to provide one if they need throttling control.